src/features/auth/signup/useBeneficiaryValidationNavigation.ts

```diff
@@ -1,45 +1,86 @@
 import { useCallback } from 'react'
+import { useQuery } from 'react-query'
 
 import { api } from 'api/api'
 import { NextSubscriptionStepResponse, SubscriptionStep, MaintenancePageType } from 'api/gen'
 import { UserProfiling } from 'features/auth/signup/UserProfiling'
 import { navigateToHome } from 'features/navigation/helpers'
-import { UseNavigationType } from 'features/navigation/RootNavigator'
+import { RootNavigateParams, UseNavigationType } from 'features/navigation/RootNavigator'
+import { homeNavConfig } from 'features/navigation/TabBar/helpers'
 import { UserProfilingError } from 'libs/monitoring/errors'
+import { QueryKeys } from 'libs/queryKeys'
+
+type NextStepNavConfig =
+  | {
+      screen: RootNavigateParams[0]
+      params?: RootNavigateParams[1]
+    }
+  | undefined
+
+export const useNextSubscriptionStep = (setError: (error: Error) => void) =>
+  useQuery<NextSubscriptionStepResponse | undefined>([QueryKeys.NEXT_SUBSCRIPTION_STEP], () => {
+    try {
+      return api.getnativev1subscriptionnextStep()
+    } catch (e) {
+      e instanceof Error && setError(e)
+      return undefined
+    }
+  })
 
 export const useBeneficiaryValidationNavigation = (setError: (error: Error) => void) => {
   const navigateToNextStep = useNavigateToNextSubscriptionStep(setError)
+  const { data: nextSubscriptionStep } = useNextSubscriptionStep(setError)
+  const nextStepNavConfig = nextSubscriptionStep
+    ? getNavConfigForNextSubscriptionStep(nextSubscriptionStep)
+    : undefined
 
   const navigateToNextBeneficiaryValidationStep = useCallback(() => {
-    return api.getnativev1subscriptionnextStep().then(navigateToNextStep).catch(setError)
+    return navigateToNextStep(nextStepNavConfig)
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [navigateToNextStep])
 
-  return { navigateToNextBeneficiaryValidationStep }
+  return {
+    nextBeneficiaryValidationStepNavConfig: nextStepNavConfig,
+    navigateToNextBeneficiaryValidationStep,
+  }
+}
+
+const getNavConfigForNextSubscriptionStep = (
+  nextSubscriptionStep: NextSubscriptionStepResponse
+): NextStepNavConfig => {
+  const { nextSubscriptionStep: nextStep, maintenancePageType } = nextSubscriptionStep
+  if (nextStep === SubscriptionStep['phone-validation']) {
+    return { screen: 'SetPhoneNumber' }
+  } else if (nextStep === SubscriptionStep.maintenance) {
+    return {
+      screen: 'IdentityCheckUnavailable',
+      params: {
+        withDMS: maintenancePageType === MaintenancePageType['with-dms'],
+      },
+    }
+  } else if (
+    nextStep === SubscriptionStep['identity-check'] ||
+    nextStep === SubscriptionStep['profile-completion'] ||
+    nextStep === SubscriptionStep['honor-statement']
+  ) {
+    return { screen: 'IdentityCheckStepper' }
+  } else if (nextStep === SubscriptionStep['user-profiling']) {
+    return undefined
+  } else {
+    return { screen: homeNavConfig[0], params: homeNavConfig[1] }
+  }
 }
```

Comment:
Cette modif va nous permettre de facilement récupérer les valeurs à passer à la prop `to` dans le parcours d'inscription